### ISSUE TITLE
Firefoxs's unity launcher: Mozilla Firefox and Firefox developer edition 's launcher "got merged"

### ISSUE THREAD

**Issue by cm-t** [ubuntu/ubuntu-make#78]
I have installed Firefox Develeper Edition From umake.
I have then a pinned launched for *Firefox for developer*.
I have also a pinned launcher for *Mozilla Firefox*.
I have opened 2 *Firefox for developer* windows and 1 *Mozilla Firefox* window (the windows you see in the screenshot).

As you can see on the launcher on this screenshot, there are little white thing on the *Firefox for developer* on its launcher icon. The expected result should be only 2 little line on  *Firefox for developer* launcher, not 3 on the *Firefox for developer*.

**TL;DR**: When opening a *Mozilla Firefox* windows, it will be stacked in the *Firefox for developer* in the (unity) launcher.

I don't know if this bugs comes from a wrong .desktop settings from the ubuntu-make package or if it comes from unity (I have seen this bug with non umake like *gnome-terminal* and *byobu*)

![Screenshot](https://cloud.githubusercontent.com/assets/2675615/6730050/cfd621c8-ce38-11e4-8b79-4989b08d5b9d.png)

![note](https://cloud.githubusercontent.com/assets/2675615/6730115/8a9bf348-ce39-11e4-8c45-4f61cfb6c8b1.png)

**Comment by didrocks** [comment]
I can't reproduce this issue in current vivid for a normal use case. I tried to (with launchers pinned):
1. start firefox dev first
2. start firefox
-> the right launcher is attributed to the right launchers

I then tried to start additional windows with File -> new window, and the right instances are set on the expected launchers.
I also tried starting then (once those instances are launched.

However, here is a use case which is failing (with no window opened at all):
1. start firefox dev edition from the icon
2. start firefox classic from the terminal
-> the firefox classic is matched with the firefox dev edition.
There is nothing that can be done in this use case, as the WM_CLASS set by firefox dev edition is:
'WM_CLASS(STRING) = "Navigator", "Firefox"'

It seems the mozilla guys need to add a special WM_CLASS to differentiate firefox and firefox dev edition. Maybe open a bug here? There is nothing we can really do on our side (being Unity/Bamf or Ubuntu Make). Closing as invalid (for this project), thus.

Hopefully, the way unity handles it, outside if you start a first window of any flavor through the terminal, enabling us to lower this wrong matching case.

**Comment by cm-t** [comment]
>I can't reproduce this issue in current vivid for a normal use case. I tried to (with launchers pinned):
1. start firefox dev first
2. start firefox
-> the right launcher is attributed to the right launchers

Strange, I tryed it again last night not using terminal (I didn't use terminal when having the bug, the launcher just got pinned up), all was correct, didn't has the issue. Did you try too with private browsing windows ?

**Comment by didrocks** [comment]
I only got the issue in the terminal case I was mentioning. in the previous post. Not sure how you got it from the launcher, keep me posted!

**Comment by cm-t** [comment]
Ok, I just coming back at home :  

 * Right click on the Web browser *Mozilla Firefox*'s launcher
 * Select Private browsing
 * bug triggered (The little white thing appear in *Firefox Developer Edition* launcher instead)
 * Moreover, other Windows from *Mozilla Firefox* will be opened on the wrong launcher after that

**Comment by didrocks** [comment]
Ok, confirmed. This is an usptream issue, they need to give different WM_CLASS hints on firefox dev. Do you mind reporting it there? Closing it on ubuntu-make for now.

**Comment by cm-t** [comment]
Link to the upstream bug report : 

 * https://bugzilla.mozilla.org/show_bug.cgi?id=1151753

**Comment by HarlemSquirrel** [comment]
I hate to bump an old issue here but this seems to be resolved as of v 58.0b1 and the bug report referenced above indicates this as well. 

I've updated my `.desktop` shortcut... 

```
StartupWMClass=Firefox Developer Edition
```

and viola!

![screenshot from 2017-11-08 09-58-52](https://user-images.githubusercontent.com/6445815/32555702-7e3fe494-c46b-11e7-9a1c-911c060e2afe.png)

**Comment by LyzardKing** [comment]
@HarlemSquirrel 
Thanks for the update! I'll change the desktop launcher that umake crates!

**Comment by HarlemSquirrel** [comment]
@LyzardKing that was a quick response! I just opened a PR for that if it helps.

**Comment by LyzardKing** [comment]
Merged!
